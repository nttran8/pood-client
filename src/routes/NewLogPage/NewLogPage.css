.NewLogPage {
  padding-top: 15px;
}

.NewLogPage > h2 {
  color: var(--color-comp-w);
}

@media only screen and (min-width: 567px) {
  .NewLogPage > h2 {
    color: var(--color-triad-b);
  }
}
@media only screen and (min-width: 768px) {
  .NewLogPage {
    padding-top: 30px;
  }

  .NewLogPage > h2 {
    font-size: 30px;
  }
}