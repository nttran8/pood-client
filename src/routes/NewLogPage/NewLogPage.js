// Libraries
import React, { Component } from 'react';

// Component
import LogForm from '../../components/LogForm/LogForm';

export default class NewLogPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };

  state = {
    logDate: Date.now()
  };

  handleAddRedirect = () => {
    // Render dashboard if add is successful
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/dashboard';
    history.push(destination);
  };

  render() {
    return (
      <section className='NewLogPage'>
        <h2>
          Log the Log
        </h2>
          <LogForm onSuccess={this.handleAddRedirect}/>
      </section>
    );
  }
}
