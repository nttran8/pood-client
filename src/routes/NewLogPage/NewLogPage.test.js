import React from "react";
import TestRenderer from "react-test-renderer";
import NewLogPage from "./NewLogPage";
import moment from "moment";

it(`Renders NewLogPage UI as expected`, () => {
  const tree = TestRenderer.create(<NewLogPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
