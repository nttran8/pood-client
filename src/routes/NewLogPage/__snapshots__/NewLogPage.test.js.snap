// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NewLogPage UI as expected 1`] = `
<section
  className="NewLogPage"
>
  <h2>
    Log the Log
  </h2>
  <form
    className="LogForm"
    onSubmit={[Function]}
  >
    <p
      className="today"
    >
      April 14th, 11:49 pm
    </p>
    <div
      className="inputForm dataBox"
    >
      <label
        className="logLabel"
        htmlFor="nickname"
      >
        Nickname*
      </label>
      <textarea
        className="Textarea nickname"
        id="nickname"
        placeholder="Minnie the poo"
        required={true}
      />
    </div>
    <div
      className="dataBox"
    >
      <label
        className="logLabel"
        htmlFor="style"
      >
        Appearance*
        <span
          className="styleDetail"
        >
          
        </span>
      </label>
      <div
        className="logSelection"
      >
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="1"
          />
          <img
            alt="type 1: separate hard lumps"
            src="type1.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="2"
          />
          <img
            alt="type 2: lumpy and sausage like"
            src="type2.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="3"
          />
          <img
            alt="type 3: sausage shape with cracks"
            src="type3.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="4"
          />
          <img
            alt="type 4: smooth soft sausage"
            src="type4.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="5"
          />
          <img
            alt="type 5: soft blobs with clear edges"
            src="type5.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="6"
          />
          <img
            alt="type 6: mushy consistency"
            src="type6.svg"
          />
        </label>
        <label>
          <input
            name="style"
            onChange={[Function]}
            type="radio"
            value="7"
          />
          <img
            alt="type 7: liquid consistency"
            src="type7.svg"
          />
        </label>
      </div>
    </div>
    <div
      className="dataBox"
    >
      <label
        className="logLabel"
        htmlFor="color"
      >
        Color*
      </label>
      <div
        className="color logSelection"
      >
        <label>
          <input
            name="color"
            type="radio"
            value="gray"
          />
          <img
            alt="gray"
            src="gray.svg"
          />
        </label>
        <label>
          <input
            name="color"
            type="radio"
            value="red"
          />
          <img
            alt="red"
            src="red.svg"
          />
        </label>
        <label>
          <input
            name="color"
            type="radio"
            value="yellow"
          />
          <img
            alt="yellow"
            src="yellow.svg"
          />
        </label>
        <label>
          <input
            name="color"
            type="radio"
            value="green"
          />
          <img
            alt="green"
            src="green.svg"
          />
        </label>
        <label>
          <input
            name="color"
            type="radio"
            value="brown"
          />
          <img
            alt="brown"
            src="brown.svg"
          />
        </label>
        <label>
          <input
            name="color"
            type="radio"
            value="black"
          />
          <img
            alt="black"
            src="black.svg"
          />
        </label>
      </div>
    </div>
    <div
      className="dataBox"
    >
      <label
        className="logLabel"
        htmlFor="amount"
      >
        Amount*
      </label>
      <div
        className="amount logSelection"
      >
        <input
          id="little"
          name="amount"
          type="radio"
          value="little"
        />
        <label
          htmlFor="little"
        >
          not much
        </label>
        <input
          id="normal"
          name="amount"
          type="radio"
          value="normal"
        />
        <label
          htmlFor="normal"
        >
          normal
        </label>
        <input
          id="a lot"
          name="amount"
          type="radio"
          value="a lot"
        />
        <label
          htmlFor="a lot"
        >
          so much
        </label>
      </div>
    </div>
    <div
      className="inputForm dataBox"
    >
      <label
        className="logLabel"
        htmlFor="note"
      >
        Note
      </label>
      <textarea
        className="Textarea note"
        id="note"
        placeholder="Clean wipe and no smell - score!"
      />
    </div>
    <button
      className="button logButton"
      type="submit"
    >
      Add
    </button>
    <button
      className="button logButton"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </form>
</section>
`;
